From patchwork Mon Sep  3 10:01:53 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585733
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E4B9C14E0
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:19:06 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id CCA0729437
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:19:06 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id BF33229472; Mon,  3 Sep 2018 10:19:06 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 281D629437
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:19:06 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=8Vyd0Pxy3YhdmWVtxSWXAKQjNZf5ovNhgG4+xlAW6lM=; b=uPmcOoKC+TOrHya1NlB3+M3IQr
	grAxefG0BVgnQlOLV4J4OSpd92azkJm2RTjnf6JpO+XLzUCVJ+5Xwxorz0GAXtQE+PLY36F9nQYkP
	L3q249EVwRR2Xy/5a72DRw5eZIlvNTm9Yht4lbz0xQ52rLpZUSsJnzWvrUPtgYTqibJ4+6bVj9YoB
	gb0XqoYnyNYlsJRELmtADgjgxdvUtimaxNT30VRn5KU9zTrkTbXD8dah2NZFQ8NosT9fTG5xMCXIQ
	W9gZYvUvYqp5CgMRBMporOADICG03xNcCTh4nF3vjWvARrIEaEXdTJcwWoSbaimBmj94RuYLBiuY/
	B0qt0VIA==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlwb-00085k-1Y; Mon, 03 Sep 2018 10:18:53 +0000
Received: from casper.infradead.org ([2001:8b0:10b:1236::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlvP-0007OZ-VT
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:17:40 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=casper.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=k6iCbW8J+KgHVaSl644rAO8j84rXqhGHUryoNJ+DbzA=; b=THxTFlahMBnGNqBrlEH7hFD/l
 LjcujtNLfNPebBmHeuXt0x6LBYd6URF87wDG8hq3v389JWsIIAMsOvZy9Wh+t3eH8YgA6J3v/KC8c
 9DUUWkIh647icETw8AZpDR7ME91Rjz6iEBhgA8mJIW1hHKdZn+kxC5wmxIiBUHStPk/P7SVQfsi2i
 BUIOFwYlNAp3MOiy0x00WmWtsSN2dQqCIbgDdstzkWGNHen24v0nFQk265VQumOoe2iJfxYKtsX1q
 yzxGg0wFizx97oR/cnqHVaXbJvNQW/eigRfI6uEyqy4mR22ZDL4emSLuoH7rseEfokh3DadmYqMed
 N8HNVX2ow==;
Received: from mail-wm0-x244.google.com ([2a00:1450:400c:c09::244])
 by casper.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgZ-0000ki-EO
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:20 +0000
Received: by mail-wm0-x244.google.com with SMTP id t25-v6so525699wmi.3
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:08 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=k6iCbW8J+KgHVaSl644rAO8j84rXqhGHUryoNJ+DbzA=;
 b=Ms/AL1dbSwu/6OkcCoXkniJHuTppTg1eS/iAfsbHlJ4Igfren9etlHtyrmUz46G8v1
 m5sOK4cOjhvE5gmLsm4B/yx/f3y8G9pf6oL9/JibL4a3Uw+ivm0U1VeQb9/JdawLRP3t
 23ryPujptKghn7ly2mJ2FmLVXW2Mxm8oPKa9RJPLyvLxOH6wBNFC70TDFW7Nu/P1qD5G
 6r2iRj8e7ZLKZKHgtqt/pW9+W+pC4jH+X9QgbL0W5eUesmDNFFOqyj3v5hCkdnu1DwYN
 ovhbadxfujnJSB6IHyflzPjmAvD/b140Z24hhQx+VgQU83T/MPPI01E0V3h6dBo1u5wv
 YOXQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=k6iCbW8J+KgHVaSl644rAO8j84rXqhGHUryoNJ+DbzA=;
 b=hMQozMDSzZQnwt5md6CNRcuwT3cT1k0dhlH2K3wq7k02+x/aFJn4T9SyvKnVhcqkwQ
 GAWpFLxSS7p58J6HzJmDQdaN4FF8nPzOjUhFRViXkO7KI//hACK1LvFHBjADtWdTsgn/
 gZ9MvJk9UfbBVaUE6MxUj0w3BpCMJEU0Uo6JZKun/hfGgV0zp+x9+1Xfv85u/w0pi8ql
 sC6Flm7IwW1Xwcw4swXLfn5wX6pUD+YUaL+yILiMNsC52wlVLIVIRbhF04oD0dKckpLF
 bbwZol4sekSroF1pG4nPIZ82HaOVLPTKvZAvtowLA27+QJlot9qO0xkHw0V6f0a3ZzAr
 WRCA==
X-Gm-Message-State: APzg51DfDmZn7Q04fYcXVkeQmudjCxyhL4hhhDmo4G1ZEyFKPpnJr0cE
 kxuU0aZeTnDjaVfDuLf2f/8=
X-Google-Smtp-Source: 
 ANB0VdYyEQP9p7oEOTJkZmuWF9IdDN1AzCi5TwwUnzPJub8vTSwG4ac49aFeYlkOajQT/IDi45yZLg==
X-Received: by 2002:a1c:8145:: with SMTP id
 c66-v6mr3739310wmd.139.1535968927560;
 Mon, 03 Sep 2018 03:02:07 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.06
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:06 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 1/9] dt-bindings: ata: ahci-platform: fix indentation of
 target-supply
Date: Mon,  3 Sep 2018 12:01:53 +0200
Message-Id: <20180903100201.23131-2-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_110219_476864_5A22D228 
X-CRM114-Status: GOOD (  14.22  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch fix the indentation of target-supply's ':'.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 Documentation/devicetree/bindings/ata/ahci-platform.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/ata/ahci-platform.txt b/Documentation/devicetree/bindings/ata/ahci-platform.txt
index 5d5bd456d9d9..b88820b4c01e 100644
--- a/Documentation/devicetree/bindings/ata/ahci-platform.txt
+++ b/Documentation/devicetree/bindings/ata/ahci-platform.txt
@@ -47,7 +47,7 @@ Sub-nodes required properties:
 - reg		    : the port number
 And at least one of the following properties:
 - phys		    : reference to the SATA PHY node
-- target-supply    : regulator for SATA target power
+- target-supply     : regulator for SATA target power
 
 Examples:
         sata@ffe08000 {
From patchwork Mon Sep  3 10:01:54 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585711
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 0A5045A4
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:10:11 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E5C0F29546
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:10:10 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id D94342954E; Mon,  3 Sep 2018 10:10:10 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 449A429546
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:10:10 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=TIP5RejUcvXXBSt3+PAt52Ykhoi1X4JBli+2a3xeEoM=; b=ggdVhsSJVQ6ruZdxWqo2kren7M
	nEwY2CpWV5myu6iMVozXEF1dOZeXvz3sUMB3NzZD/HMP86h6g9F/SUjKVekMHaQzYrRkHmFytmn4/
	IwfHwAlJMp701gcFXvAZtAn18enahtHtRZYZOuUfm2I0p3v2y1cPlcqGDaHlFIT1d0h1nAorrbedX
	m+1HGUp2aQhYwbj5vQWGfPw3EMdKJ0R7yUo4Cbtw5A0LW9UYddNGMYwMwtzvKIyneUKfwrDhbjLNh
	odHSVJbkcoA+nZpOZ2VfBjIwjyHb8aKwjlI3KChMAnAMp9MZr/SBN/IpSxeh7OOmYiPPYdn4fIUUi
	dyOs+vTA==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlnv-00029W-GB; Mon, 03 Sep 2018 10:09:55 +0000
Received: from merlin.infradead.org ([2001:8b0:10b:1231::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgk-0006c6-Ul
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:02:31 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=merlin.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=t7Yy96h1CNLQsmpPB+gBtt+3MwMJX3o+4LyGaKYq94s=; b=XMc7NKu6722+VUw22ZyoP0YUW
 /6NyM188BStMRUzc9iA19UIDUVrt9RS97PuGfblHUr0ZTKDlioLv1Kg36c27BY7qkBy8JUk1xt36x
 6VHEiTX1BTC06beEMELHX5iMVMSEzJ5iRZAWmRyWEbZWrCg3JUCk0WYisfMM2AiM+nJpLgLI16mwG
 7kLy9No1MttRQ+HqCiJuvohjvlzQIJzINT1Np6C+QIM/ej7kkWe9mgQ6rkjRY9ygBFRwSOAOKkgLq
 QA3nAf3xZsRHuuB3P7R6/5TIaBJJrUw0RVwDsyw24QsEJQSFZQYdaHfL2CJqEM+hOSKa5/l7gmre3
 ukVyIHcnw==;
Received: from mail-wm0-x244.google.com ([2a00:1450:400c:c09::244])
 by merlin.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgd-00070y-7t
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:29 +0000
Received: by mail-wm0-x244.google.com with SMTP id t25-v6so525810wmi.3
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:11 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=t7Yy96h1CNLQsmpPB+gBtt+3MwMJX3o+4LyGaKYq94s=;
 b=m070xjmTV8oB18DF8XHLMSb11OXMKQs4HbkTJBzGsA/kkCepsrdAH/620RDjGWUxkr
 fVxM32Xk6O2ToADYUpC/Bq1fbBsqzz92WEv6rSx+LGrz9kYfmE73CQe85+DXUqdmUBaE
 uWoKO6TxQ4k8oheINrk/Jq9Ac6XrtY5d2/O2jLgsq96FdD8xXuF2Nl9aLvUQghs7ofoe
 nUoBZa99VKNQ+jcvmjSFELfUQbPnZ9lmh/l7UQq3nnfTTZ7kPUdp65SqiBqvf8gtq69B
 YqDJdpoLk/EmwXZcd40Kii785c1BHBSZ6eKnbDhQ+TROcSG2r8uN4HXB0TjMrtLBAFWt
 4xwA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=t7Yy96h1CNLQsmpPB+gBtt+3MwMJX3o+4LyGaKYq94s=;
 b=Ka7CglbNhGFqJau1GOCE/wkSoqHPqWKrvyZlQEmw1fmuMG/LnKa7Rx2hblBOCMb83e
 8jZqMs4mUi4J9XCq+WU+Y6SEnF1NTEHOzx5aQfMtlSVEgty5W3y3WQgnehDz7Jtxcw9N
 bpxmiqDrnxRsIzCLUDKZFYzgJz0NCEtWB9YBCV4UecfP0QMfHxHKd/gPi/u4lrp50nKm
 XNuWjeXfEbvhasiShREXQqPTlIsXBXwSMg9EkxRVSw7MaNcXBXMgCjhyOhH4uJbptkLU
 t1fgN8hrnXbw8hfJjtI73jiJ5WTV05zTB3n2kUs+xBlDQKcDa5tUl/uB9PQdsqJ9tj0L
 /lZQ==
X-Gm-Message-State: APzg51D+xStAb17vYk57EIG5i7oRNYWIHVR+5tXqsWGQbO5DnHsCn7GK
 /3PitaLTWdL9nLDn2vpBFTY=
X-Google-Smtp-Source: 
 ANB0VdaOCrX22FEezBJWtRmBN94RtIe219A0b5ROX8xCmEYZdUsBkS2c17wZUAYKwtPmJevJ1M10EQ==
X-Received: by 2002:a1c:1d87:: with SMTP id
 d129-v6mr4492426wmd.34.1535968928948;
 Mon, 03 Sep 2018 03:02:08 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.07
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:08 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 2/9] ata: ahci_platform: add support for AHCI controller
 regulator
Date: Mon,  3 Sep 2018 12:01:54 +0200
Message-Id: <20180903100201.23131-3-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_060223_318444_F179A830 
X-CRM114-Status: GOOD (  22.00  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

The SoC R40 AHCI controller need a regulator to work.
So this patch add a way to add an optional regulator on AHCI controller.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 drivers/ata/ahci.h             |  1 +
 drivers/ata/libahci_platform.c | 26 ++++++++++++++++++++++++--
 2 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/drivers/ata/ahci.h b/drivers/ata/ahci.h
index 6a1515f0da40..1415f1012de5 100644
--- a/drivers/ata/ahci.h
+++ b/drivers/ata/ahci.h
@@ -352,6 +352,7 @@ struct ahci_host_priv {
 	struct clk		*clks[AHCI_MAX_CLKS]; /* Optional */
 	struct reset_control	*rsts;		/* Optional */
 	struct regulator	**target_pwrs;	/* Optional */
+	struct regulator	*ahci_regulator;/* Optional */
 	/*
 	 * If platform uses PHYs. There is a 1:1 relation between the port number and
 	 * the PHY position in this array.
diff --git a/drivers/ata/libahci_platform.c b/drivers/ata/libahci_platform.c
index c92c10d55374..a886b61476a3 100644
--- a/drivers/ata/libahci_platform.c
+++ b/drivers/ata/libahci_platform.c
@@ -139,7 +139,7 @@ EXPORT_SYMBOL_GPL(ahci_platform_disable_clks);
  * ahci_platform_enable_regulators - Enable regulators
  * @hpriv: host private area to store config values
  *
- * This function enables all the regulators found in
+ * This function enables all the regulators found in controller and
  * hpriv->target_pwrs, if any.  If a regulator fails to be enabled, it
  * disables all the regulators already enabled in reverse order and
  * returns an error.
@@ -151,6 +151,12 @@ int ahci_platform_enable_regulators(struct ahci_host_priv *hpriv)
 {
 	int rc, i;
 
+	if (hpriv->ahci_regulator) {
+		rc = regulator_enable(hpriv->ahci_regulator);
+		if (rc)
+			return rc;
+	}
+
 	for (i = 0; i < hpriv->nports; i++) {
 		if (!hpriv->target_pwrs[i])
 			continue;
@@ -167,6 +173,8 @@ int ahci_platform_enable_regulators(struct ahci_host_priv *hpriv)
 		if (hpriv->target_pwrs[i])
 			regulator_disable(hpriv->target_pwrs[i]);
 
+	if (hpriv->ahci_regulator)
+		regulator_disable(hpriv->ahci_regulator);
 	return rc;
 }
 EXPORT_SYMBOL_GPL(ahci_platform_enable_regulators);
@@ -175,7 +183,8 @@ EXPORT_SYMBOL_GPL(ahci_platform_enable_regulators);
  * ahci_platform_disable_regulators - Disable regulators
  * @hpriv: host private area to store config values
  *
- * This function disables all regulators found in hpriv->target_pwrs.
+ * This function disables all regulators found in hpriv->target_pwrs and
+ * AHCI controller.
  */
 void ahci_platform_disable_regulators(struct ahci_host_priv *hpriv)
 {
@@ -186,6 +195,9 @@ void ahci_platform_disable_regulators(struct ahci_host_priv *hpriv)
 			continue;
 		regulator_disable(hpriv->target_pwrs[i]);
 	}
+
+	if (hpriv->ahci_regulator)
+		regulator_disable(hpriv->ahci_regulator);
 }
 EXPORT_SYMBOL_GPL(ahci_platform_disable_regulators);
 /**
@@ -351,6 +363,7 @@ static int ahci_platform_get_regulator(struct ahci_host_priv *hpriv, u32 port,
  *
  * 1) mmio registers (IORESOURCE_MEM 0, mandatory)
  * 2) regulator for controlling the targets power (optional)
+ *    regulator for controlling the AHCI controller (optional)
  * 3) 0 - AHCI_MAX_CLKS clocks, as specified in the devs devicetree node,
  *    or for non devicetree enabled platforms a single clock
  * 4) resets, if flags has AHCI_PLATFORM_GET_RESETS (optional)
@@ -408,6 +421,15 @@ struct ahci_host_priv *ahci_platform_get_resources(struct platform_device *pdev,
 		hpriv->clks[i] = clk;
 	}
 
+	hpriv->ahci_regulator = devm_regulator_get_optional(dev, "ahci");
+	if (IS_ERR(hpriv->ahci_regulator)) {
+		rc = PTR_ERR(hpriv->ahci_regulator);
+		if (rc == -EPROBE_DEFER)
+			goto err_out;
+		rc = 0;
+		hpriv->ahci_regulator = NULL;
+	}
+
 	if (flags & AHCI_PLATFORM_GET_RESETS) {
 		hpriv->rsts = devm_reset_control_array_get_optional_shared(dev);
 		if (IS_ERR(hpriv->rsts)) {
From patchwork Mon Sep  3 10:01:55 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585705
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 409B0920
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:07:51 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 28C7828D95
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:07:51 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 1B39A294CC; Mon,  3 Sep 2018 10:07:51 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 81C7728D95
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:07:50 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=KUT7WohEVIEzdcHRxg6hmurWE3ji7Ahj7qIMhT8K3go=; b=P9twkAZWt4rxAa31hUq2FKnpt5
	SaqH+TkndYg1EjCxstPUI+Oo/xKFIqywRltRnBZ8s30pakGI5E60RxzQ80Ih7kFBnYhCsK3Zctizp
	Gl+ypa3ioTq7m1PGAToCh/rNg6P7jcYCYA9E8GGY54VBKMg1Vk/nk6aUAv5I/V2BbA0nMDX197Hhn
	Es+I5tnjL+gY73b5mwofRi9TYSdb5rItpyDCAZM9xAunAGQPMtsAJv9xDtcV83uY/E2K5z3QsoG8s
	mL5rfRpZ9LPz9qM2r22+scDo+zDtuAlPpGQvgJLtBEEd0TcsfQQPO1CQtlKe/QM5fp/ph5XgcR56Q
	CM3uUdQg==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlli-00015M-Dp; Mon, 03 Sep 2018 10:07:38 +0000
Received: from merlin.infradead.org ([2001:8b0:10b:1231::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgh-0006ai-Ll
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:02:27 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=merlin.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=/+HNfr3I0MzGHGd7Aj3LN1rm33wfTbHxswxtDehlIDw=; b=sEO5GXJal/V+vKvpM2/WVZl1N
 tO2aEI+0ZbRtztJ+LcMa7c8C8mdoPYSUgJwY2Dp6RfaRXCX+djcHrfXuaAyqcGUr2ELKot+ZtKf6b
 jjgnPIR4vpYzXdLqv1co0Y61N/fGsNU0FHZ/C6bH7kJKqYSirIcUXok+Ql85hsA2oroXIW1LIw+R5
 +bjWBi44OXML1340xuJCqNHCx4c44X1dkkbjil4d6hfaLrA4nyky1xMVk+NzxQGu6XxU9Npe5D3MI
 CAKvVSkKL8syspsvcRgMVx2oOWW/qyQ40z/tr2xWM+7WPhvE27ppu9aEd3Fgyl0ULanii2yrslvz4
 EytwkMugg==;
Received: from mail-wm0-x241.google.com ([2a00:1450:400c:c09::241])
 by merlin.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgd-000710-AE
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:25 +0000
Received: by mail-wm0-x241.google.com with SMTP id s12-v6so563809wmc.0
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:11 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=/+HNfr3I0MzGHGd7Aj3LN1rm33wfTbHxswxtDehlIDw=;
 b=gEDqDcKo7iy9T0WsJtmQLxyIP9rtZgMc9YreV4LnDm2vMI3qxoT5/qt/KAPFnFYhuP
 by+38DFjwWf5t0BKdPZ8F2FO65Cq8ZBga4J9kyowCpuGj38qDMS00JgKGm260tXMjzsC
 fVkYwl2A4C5aC6HweijBzuHZmwYUHJ6bASLi/FuA4pGFDn9zxrBQSICcVZLJ7tJyv0yC
 J4sTKmcr4xlRZA+ikai0huBh19pO7bTvyP7juqoyswIusWzsXZA2f8GAd7IP5oOc+eiF
 y6MxfhcAW/Z0hAKMYCVm93mWAXR0y6sf8Lro5eNagQ1vD8hlLqwPVPh4KOLmuP5Gs6Gq
 W7Yg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=/+HNfr3I0MzGHGd7Aj3LN1rm33wfTbHxswxtDehlIDw=;
 b=hluOCnkjD9PWRbCjECYbcmsLyQ1ju5AyTsY0RMRMFz0gsn8CdAutfHpdyaBHeI1Ry9
 oKlGU4AR1UbSG5E8EmsJyly03ilTt7T2u0EwK0eQ+kWXiQwPM0j5kIAX8q6VG35+VWlf
 Nh1BeTV3aY8rhM0Hmm7mkW2J/LIiOCFqtcn+4VYLi/Le/zxLNqboCSpceOKuFmKuYz9e
 BUIgf9y7Xu6LLtap3LmZYxMO+qXAY1qEis9nngWDvgF0J55BkKqxtKXqTks0n/971s28
 TJ/wW6fGXppGiSY9NU/sO9YXTEISVA/0WBcmwSPTxjXUZZXQ6CjgOdjLiyNqGYSrTYvT
 NQfA==
X-Gm-Message-State: APzg51Anh6wKENkgelhXlPiLcTr3KhUsIhywX4puFkQZ3ESO0GFB/THf
 xCH2fdpcMoCFk3kWVG42hpw=
X-Google-Smtp-Source: 
 ANB0VdYhlcIdxIk13NqnuVtKSdI2MnJEZ1y776Cx47R3Q6YYuWJ4e0r+t4sXdpC8xJ31nTgyCLrO5Q==
X-Received: by 2002:a1c:e70b:: with SMTP id
 e11-v6mr4496423wmh.22.1535968930273;
 Mon, 03 Sep 2018 03:02:10 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.08
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:09 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 3/9] dt-bindings: ata: ahci-platform: document ahci-supply
Date: Mon,  3 Sep 2018 12:01:55 +0200
Message-Id: <20180903100201.23131-4-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_060223_447177_EE4E87EB 
X-CRM114-Status: GOOD (  13.57  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch document the new optional ahci-supply.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
Reviewed-by: Rob Herring <robh@kernel.org>
---
 Documentation/devicetree/bindings/ata/ahci-platform.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Documentation/devicetree/bindings/ata/ahci-platform.txt b/Documentation/devicetree/bindings/ata/ahci-platform.txt
index b88820b4c01e..f495774c8af9 100644
--- a/Documentation/devicetree/bindings/ata/ahci-platform.txt
+++ b/Documentation/devicetree/bindings/ata/ahci-platform.txt
@@ -33,6 +33,7 @@ Optional properties:
 - target-supply     : regulator for SATA target power
 - phys              : reference to the SATA PHY node
 - phy-names         : must be "sata-phy"
+- ahci-supply       : regulator for AHCI controller
 - ports-implemented : Mask that indicates which ports that the HBA supports
 		      are available for software to use. Useful if PORTS_IMPL
 		      is not programmed by the BIOS, which is true with
From patchwork Mon Sep  3 10:01:56 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585717
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id DB827920
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:12:20 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C02E029560
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:12:20 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id B196629567; Mon,  3 Sep 2018 10:12:20 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 1E30B29560
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:12:20 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=RKP6IxqRfoPA9NtrC9RfjywCEDZLZF+aVxQ6XMjuKq0=; b=LuyNlgalWAq1VJVbsflih7GsV5
	BzAls8s8vArjnuVFR8QA2bw6d5PEpJwO9zOJan2VbNnbZcIOazxssI5WrzvA0R1uJOgG5YGwRCi1G
	g2RKjE13k8dojOjXe276lVU+M2uY6OO9U8ktskbbx1n0Q36X2/nbv3TfnKgyqLHqzplVuc3oGSEf+
	Y6iI3Gz4ylAa1Uq0DqoWuRtUdFGJrHg+dm83tVnYGhb1RItVUi1NaNWUt5FjWCIkqwyDPUM9+efHd
	Yi6fDaELpsHAN+FIDyK5v8t099Y+iYYJAJ8N/zPlD2ihvavDGMwHd2eHmOks3vYcqrHLWPMgDo9wK
	TBtxQbMw==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlq9-0004Jv-Dn; Mon, 03 Sep 2018 10:12:13 +0000
Received: from mail-wm0-x243.google.com ([2a00:1450:400c:c09::243])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgd-0006Vd-DI
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:53 +0000
Received: by mail-wm0-x243.google.com with SMTP id y139-v6so536284wmc.2
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:12 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=9LXnz2fUikvXtzGPguMQqVne60eSM6Bi3/wfM9ERA6c=;
 b=ASc6T1XRSP/kNEywFxw90OG1YnpHJ4VQZ+PqmR+SWLpIB9Wh7pYhzG+vAE0G+Ebi1E
 O5QFT4TH9NSfb3jjSv/9OSYhX+jnrslrUWNL53S/hupKMjmjeMhpkRiT+NOv0g8lW8ph
 1i32SWXceUoP9UE1h/LfZ2HzgstomtNjVbMZ1EDd5MZGWa3Xg8/jwK2/yaD4EGQfyjMZ
 90gwkayHWTrzAnglB96xpKyFTX/JpaKlRU3DCa9mRuesMayYa0Gr3hT6TnNluM26PzCy
 XFY+IozKXf/TTz32pTRbCaF4UUR7w15OV6yp+E2G8VXuJbqvrnlntVgD76CNEgTbh151
 dBGA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=9LXnz2fUikvXtzGPguMQqVne60eSM6Bi3/wfM9ERA6c=;
 b=DlcrqC4qovBAWJABXE02PqONNr1JyPMq0yBKV+bfG7biZO/jQ+gVILA5QdoTCYXK5p
 d7zfVdcGfDt6CCg+nj8hHJh+zzdWWqYPz6L5UCoGzxJmdN5f37qh63EJpT50XYeYnW9A
 SGOwhXM5vobgO4nG/r8OKvGteJ6MJxakEaQRxW+uB/OH5NCaW2O0ZpLdiwPeJfO5o39X
 F47NE6/TZqM9r55eCTwZA1qiceOChEXW3FNWO5rlwcIHEIN4IAR90lZUK9u9SBV0suS8
 n68dysooX2ZU/1nAaiE4IKQ58v7gY3Em3Qq3r3YpsVKnQBFD8W0HP58kBTUb1Awr/+Lw
 s2Rw==
X-Gm-Message-State: APzg51CcJjceEVOycHVrZwMfETDsefkcVR2+8C/h2XeYds2vZszoaElU
 nS9u57CpCXZaOItpqaHqb2k74O5C
X-Google-Smtp-Source: 
 ANB0VdZI120x0VHUXWNAWmRFu+EcW5fHDIy9v8o9znsWpScJc35G4xbJN0WpuA+55izJLaYmVUDo5Q==
X-Received: by 2002:a1c:32c4:: with SMTP id
 y187-v6mr4447461wmy.31.1535968931627;
 Mon, 03 Sep 2018 03:02:11 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.10
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:10 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 4/9] ata: ahci_platform: add support for PHY controller
 regulator
Date: Mon,  3 Sep 2018 12:01:56 +0200
Message-Id: <20180903100201.23131-5-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_030223_585938_8ADFE36D 
X-CRM114-Status: GOOD (  16.49  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

The SoC R40 AHCI controller need a PHY regulator to work.
But since the PHY is embedded in the controller, we cannot do a DT node for it,
since phy-supply works only in node with a PHY compatible.
So this patch adds a way to add an optional phy-supply regulator on AHCI controller node.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 drivers/ata/ahci.h             |  1 +
 drivers/ata/libahci_platform.c | 20 ++++++++++++++++++++
 2 files changed, 21 insertions(+)

diff --git a/drivers/ata/ahci.h b/drivers/ata/ahci.h
index 1415f1012de5..ef356e70e6de 100644
--- a/drivers/ata/ahci.h
+++ b/drivers/ata/ahci.h
@@ -353,6 +353,7 @@ struct ahci_host_priv {
 	struct reset_control	*rsts;		/* Optional */
 	struct regulator	**target_pwrs;	/* Optional */
 	struct regulator	*ahci_regulator;/* Optional */
+	struct regulator	*phy_regulator;/* Optional */
 	/*
 	 * If platform uses PHYs. There is a 1:1 relation between the port number and
 	 * the PHY position in this array.
diff --git a/drivers/ata/libahci_platform.c b/drivers/ata/libahci_platform.c
index a886b61476a3..dc4d79b1c9ae 100644
--- a/drivers/ata/libahci_platform.c
+++ b/drivers/ata/libahci_platform.c
@@ -157,6 +157,12 @@ int ahci_platform_enable_regulators(struct ahci_host_priv *hpriv)
 			return rc;
 	}
 
+	if (hpriv->phy_regulator) {
+		rc = regulator_enable(hpriv->phy_regulator);
+		if (rc)
+			goto disable_ahci_pwrs;
+	}
+
 	for (i = 0; i < hpriv->nports; i++) {
 		if (!hpriv->target_pwrs[i])
 			continue;
@@ -173,6 +179,9 @@ int ahci_platform_enable_regulators(struct ahci_host_priv *hpriv)
 		if (hpriv->target_pwrs[i])
 			regulator_disable(hpriv->target_pwrs[i]);
 
+	if (hpriv->phy_regulator)
+		regulator_disable(hpriv->phy_regulator);
+disable_ahci_pwrs:
 	if (hpriv->ahci_regulator)
 		regulator_disable(hpriv->ahci_regulator);
 	return rc;
@@ -198,6 +207,8 @@ void ahci_platform_disable_regulators(struct ahci_host_priv *hpriv)
 
 	if (hpriv->ahci_regulator)
 		regulator_disable(hpriv->ahci_regulator);
+	if (hpriv->phy_regulator)
+		regulator_disable(hpriv->phy_regulator);
 }
 EXPORT_SYMBOL_GPL(ahci_platform_disable_regulators);
 /**
@@ -430,6 +441,15 @@ struct ahci_host_priv *ahci_platform_get_resources(struct platform_device *pdev,
 		hpriv->ahci_regulator = NULL;
 	}
 
+	hpriv->phy_regulator = devm_regulator_get_optional(dev, "phy");
+	if (IS_ERR(hpriv->phy_regulator)) {
+		rc = PTR_ERR(hpriv->phy_regulator);
+		if (rc == -EPROBE_DEFER)
+			goto err_out;
+		rc = 0;
+		hpriv->phy_regulator = NULL;
+	}
+
 	if (flags & AHCI_PLATFORM_GET_RESETS) {
 		hpriv->rsts = devm_reset_control_array_get_optional_shared(dev);
 		if (IS_ERR(hpriv->rsts)) {
From patchwork Mon Sep  3 10:01:57 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585731
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 221E014E0
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:18:22 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 0B77228D1F
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:18:22 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id F3B8428D33; Mon,  3 Sep 2018 10:18:21 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 7F3A528D1F
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:18:21 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=SRJndIAI8DCZp5H0DQNjjYcLTyWKYMtwsGixfOf5wz0=; b=HzcHSjQWBxPHe7FpzGyHPczhBW
	wV0+J8aj84syYbOkqP5wVKnCQnASavykahlGkPrO/rsFJTcP5GP/MM4BPqZQDLQFA0Kv1msVaM61E
	AwPfI0D6zVwioc5rbq2X5LJmewoYHDUr6XmwTQEOnhLQKePjVuw0xKw7P7nfZQSLWcgVWuh6nflqH
	TiY81NDfIueJt0dRtlXVuYnaO28ygDR7GYSCc9XnWBKn1zsb8+5yHZXtvYP/L5zaLOy1VoSKzC1Dh
	aI1ucMnHXPqgVszzMChGpL9UtnYuXbXW2I6QleFU6vY29z1NX7YqpqZmTQNW6OOKjkMtP4oo5cFRD
	yS4+zaUQ==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlvt-0007mP-Rv; Mon, 03 Sep 2018 10:18:09 +0000
Received: from casper.infradead.org ([2001:8b0:10b:1236::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlvO-0007UZ-IF
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:17:38 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=casper.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=RUdmpyjvYH/P3gTVki1Kcr+96q6wklBBPgCeGs+6Dmc=; b=u6Crk4ERFvDsQrNeTCz6Sfn8g
 ah6cvDohKu7U/ri9F2dQFDl7tbJoED1Eoo5/3GMB5C9hd/PrZlJzKWvkyS4wdtXIDEtISU1Zde58r
 qEJ+F4cwnKb77M3v03aO9HWiUuY7L1KGqSYh+/vu/eVmTtu75QAcN2rAkM3fCkmUo4CcgTbzBqNID
 dxP7sI0GBw0qxb0b/1jWyYyAXhUspYnuNm7sT5JBkL6ytjjlCSVAZqS0nDIIxrjjPsA9jcZoFfMUU
 tmkOUdw1E50yHXj53hAFw+04F/XH709PmU4V+FjbvWeMXauMOo0QJiYNCsUDnepIkjfLrxnbyiaJR
 Dfqgp7ZQw==;
Received: from mail-wm0-x244.google.com ([2a00:1450:400c:c09::244])
 by casper.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgZ-0000l7-ER
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:25 +0000
Received: by mail-wm0-x244.google.com with SMTP id o18-v6so548709wmc.0
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:14 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=RUdmpyjvYH/P3gTVki1Kcr+96q6wklBBPgCeGs+6Dmc=;
 b=laD9dgftVxceZOvDHfCHlcltULXv1fqRzKjtuF6i1CuYCH/3IpLneIU0UA4HTWLzbm
 k7/KWF2eyedhidhT7l8W32vTlv/h6ybsNUqQ8XjhXKsPQIL2jwEMbZK9JkJAeP8oRTMp
 qtxIfTf4Xe9bG+2LoPGO5i7T1WKibKTBMdw7e85Dxt98zyvdOpHGBWdcZz5wEGhhxtuC
 xPt4t22anO0uGSCM0Ghc5YX9pyZzcBYVBWuBAa5UKUdfEV6eAiha9zB8YTjWTWSs1bqC
 lwdGAFQsJ1Clf7KWTH97gmoIQ6Bl6TPY+/21UW3JLhiJipizln9AfdYWnDldNzBxyFiI
 T+lA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=RUdmpyjvYH/P3gTVki1Kcr+96q6wklBBPgCeGs+6Dmc=;
 b=TLUEjcJgl9yTX+TbkL9opBmiM9d3jeIdOJohkw8kkpAm+M7b02aVGGKFeEep9UDjpZ
 +055dLgd3/Vb/AdbYfnEJ+EfOompSvGuKnuefJw1fZLU10JNA9OAa9VF3rphm2P3Rf74
 lcPemT56XTV3WgPM+syn9rPwCl3Pei8uGwAL0dX1C2a+eqMLBXAjLE+UVLde2v7nQnYF
 /5hJsd2xFm8haKAsdX8KkDmvw0XhzegPQ1YjHddrLPidw285YO/ra3Zq7CLIJxY5nA4v
 yLb76D0VKPKR8ZD7+YkxoMYRC69zxjcSfNLsADnviergrEpsxFNOXw6uxOBOfE5RMuHp
 653A==
X-Gm-Message-State: APzg51BAkQWlV12SgEvruKAhfxZr0XtN4R2vwZBitp3Jhiud1zRDcFUB
 51085nVn1VoFtzqc6S8d2L0=
X-Google-Smtp-Source: 
 ANB0VdYO7WlmRYurEh9tWRufF7iTZ5egaNo4eHQzghRCOy6uc+EMDSvuAWuAJCI1AcCDdzcCYZnuJg==
X-Received: by 2002:a1c:2e11:: with SMTP id
 u17-v6mr4638794wmu.102.1535968933093;
 Mon, 03 Sep 2018 03:02:13 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.11
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:12 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 5/9] dt-bindings: ata: ahci-platform: document phy-supply
Date: Mon,  3 Sep 2018 12:01:57 +0200
Message-Id: <20180903100201.23131-6-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_110219_483395_D528FE41 
X-CRM114-Status: GOOD (  11.82  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch document the new optional phy-supply.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 Documentation/devicetree/bindings/ata/ahci-platform.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Documentation/devicetree/bindings/ata/ahci-platform.txt b/Documentation/devicetree/bindings/ata/ahci-platform.txt
index f495774c8af9..45b451961612 100644
--- a/Documentation/devicetree/bindings/ata/ahci-platform.txt
+++ b/Documentation/devicetree/bindings/ata/ahci-platform.txt
@@ -31,6 +31,7 @@ Optional properties:
 - clocks            : a list of phandle + clock specifier pairs
 - resets            : a list of phandle + reset specifier pairs
 - target-supply     : regulator for SATA target power
+- phy-supply        : regulator for PHY power
 - phys              : reference to the SATA PHY node
 - phy-names         : must be "sata-phy"
 - ahci-supply       : regulator for AHCI controller
From patchwork Mon Sep  3 10:01:58 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585707
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1D40C920
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:08:26 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 055ED294F9
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:08:26 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id ED60B29518; Mon,  3 Sep 2018 10:08:25 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 7A01D294F9
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:08:25 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=Ilgsa+qx+k57v5ooZJeQLM+2sJTng6XA28eBx+xgdUM=; b=IRTS8aTGJNU6qDohbLXAxXEm7Q
	P6DDdumFy6vF8Pee5LM5XteMFQAG8d2MaiJmd8UDwcEQt7BGHIn7eFU3SrSSTpciQ0IKgYXgwgzEY
	ljBfv2Hw1ii6itOlbdYc3G/9Fd/+5SeBvcKJBukQB2IUI/SRwvLvkfuRvONebWjhSWHzNlQe2+IEs
	shSRGfpi9SEjB/0MUca1+oqksiS6W3bpROCtVRsA3o5elg84TAEqvJApoWYy+pfW4g7B2E9jcYWOn
	a0mVGA2Pl+/E1miqf47u7Twy1je3nqUhcx81g/9ETZFZGHQuhWvm5u0oidIbDtItRptmIRvCuKfRH
	kFWyr89A==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlmO-0001Pa-Da; Mon, 03 Sep 2018 10:08:20 +0000
Received: from merlin.infradead.org ([2001:8b0:10b:1231::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgh-0006aa-DD
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:02:27 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=merlin.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=tfuzh9T+CpyAPnemdnqeg4728XIZ30noV5zNc4YBbSs=; b=QpM7Y2jP0h7hZ5aOTrh14AOBP
 iapnO48pAEg22tW4ym6BcGjx+REHsUfJgUutGCE9KSaRcJNtq6sWDTTUQlFdxpMdbQY/ZNcweCed5
 IHeANhgQeJe9ci3VK9zLPOewBOszkifZr4CrngYGlWZ6eyX/5KvL+C9uVKCbwlE3UPkRu/I8rML5j
 TZ8rAv0nmI6YEBautOoCfZw1b6OnUWiEhhAEl/yOaS7x2wLLOBTPXqQ/o/s6z30b24U3X1R5eZhbr
 UqIcl7YhcSCTTPU654uV0QjE/29n35TJBc5Boolu6iH/MD5RcIDKyw49D3mK5eit+8ibs1G9gwQgf
 TbRj/yQow==;
Received: from mail-wm0-x244.google.com ([2a00:1450:400c:c09::244])
 by merlin.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgd-00071A-9F
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:25 +0000
Received: by mail-wm0-x244.google.com with SMTP id c14-v6so538603wmb.4
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:15 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=tfuzh9T+CpyAPnemdnqeg4728XIZ30noV5zNc4YBbSs=;
 b=dkY7GaVzP2wLdFWCwyAMlswzjqggspJYq8VFG7wQe9/IZAXX6xJqJKxB783tYp23mG
 hhrySEyAMj/TN6r9QQGrQY2JhGI1vt/BEKEl3tLe/UQjfuzy9GhfLlZCMmDvURdr9zY5
 yTDgyt/3Gov6fU4iL/zfEE6ntECK1PkvNmryp5djRcGbGcxOXxYVyoFD4J4hNk+a+ODb
 RdNHA6zniX6ngL7ElgrsP9Qb7xNWzoSXu+Zbzb08lESQ7ptWki58I/YZiG2jeH2S80G5
 HJKyNyJd9F3fUXAkyR7w/snERMA1yAZShXQi0E2JyNilVXqEWYNmzaEdufWSlVSivwSZ
 8C8w==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=tfuzh9T+CpyAPnemdnqeg4728XIZ30noV5zNc4YBbSs=;
 b=GMC3XgdfymVgn2N0pohPaxxlCd3VxjJfer+N85Bh56oTSmUDlKnyrTyf5s0Vc2ssqB
 xn5AelwHUg9rJTkwQ3W9KPbE7OFg1p/cg+lt2oXt8ly0x8jIkSL0d1bk1ZNmuk/W3Ia6
 xIS7TXyK+N+XShYhKM3/5yU7227otCV2OjvOG1viQW1nbTwniN2nZugXxkzvmAnWz67u
 5wq0SjaNDg1gIIYCMPCXEW7DxKdy7oVMj/cCkHJH6TCdDlnry/NHf92BAphZz/NacPAV
 hr6gz/8wc4Y0Wy5DSyteNR/rLEkeSqscsbVd+h+AY7LCIg0ihbLaL63Fv0Y5iq4Dmn9U
 My6w==
X-Gm-Message-State: APzg51BGlRsnyB1Gn2EWFefT1HV67k5eK2i+yQsGXzT7v9sxzYwSbtK7
 mGf2lFhTEkjri7UwwgADlpM=
X-Google-Smtp-Source: 
 ANB0VdZpQGuSBITo3ceVBcWUmYxP6UpkNxPjKYVo2cynUaeXk/8PYnZXGOxJoI86W7paaHTYauwDxA==
X-Received: by 2002:a1c:b441:: with SMTP id
 d62-v6mr4695405wmf.17.1535968934457;
 Mon, 03 Sep 2018 03:02:14 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.13
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:13 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 6/9] ata: ahci_sunxi: add support for r40
Date: Mon,  3 Sep 2018 12:01:58 +0200
Message-Id: <20180903100201.23131-7-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_060223_375544_C7FF66B6 
X-CRM114-Status: GOOD (  15.38  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch add the r40 compatible to the ahci_sunxi's supported list of
compatible.

Since R40 need ahci_platform to handle the reset controller, we also add
the new AHCI_PLATFORM_GET_RESETS flag for ahci_platform_get_resources().
This has no consequence for older platform (a10, a20) since the reset is
optional.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 drivers/ata/ahci_sunxi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/ata/ahci_sunxi.c b/drivers/ata/ahci_sunxi.c
index 631610b72aa5..911710643305 100644
--- a/drivers/ata/ahci_sunxi.c
+++ b/drivers/ata/ahci_sunxi.c
@@ -181,7 +181,7 @@ static int ahci_sunxi_probe(struct platform_device *pdev)
 	struct ahci_host_priv *hpriv;
 	int rc;
 
-	hpriv = ahci_platform_get_resources(pdev, 0);
+	hpriv = ahci_platform_get_resources(pdev, AHCI_PLATFORM_GET_RESETS);
 	if (IS_ERR(hpriv))
 		return PTR_ERR(hpriv);
 
@@ -250,6 +250,7 @@ static SIMPLE_DEV_PM_OPS(ahci_sunxi_pm_ops, ahci_platform_suspend,
 
 static const struct of_device_id ahci_sunxi_of_match[] = {
 	{ .compatible = "allwinner,sun4i-a10-ahci", },
+	{ .compatible = "allwinner,sun8i-r40-ahci", },
 	{ },
 };
 MODULE_DEVICE_TABLE(of, ahci_sunxi_of_match);
From patchwork Mon Sep  3 10:01:59 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585709
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id BBC235A4
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:09:15 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A3D5B29522
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:09:15 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 9668429536; Mon,  3 Sep 2018 10:09:15 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 254B929522
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:09:15 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=rJfWfWkk+BTpbtR7NUZUSBuRbO71yaIUPSUj377fT/A=; b=iJhCCXZIuwFYN3nFMDGtaVFKPX
	w+rvVnIw5dnyC0L8XnkxQGFt3hfsxiV7CgpAsPZQicCMQY/Qcjyw6C1g6XA3X5nfoxz9H8yq7z8ki
	v6dd/T2vePsmpItsGYotKesH1n6eY6pTxC+E/DpSNxRbF24I4r8hanVvwJ6neagyeQrzpyJFu0unV
	+jPJ2Zpu22YyFQ0I8vc9fBvsaqsgbvmGUrNVQYkqhiRdq/uVWBNRQOwnKaVsCy49asUbd4GuBQO8h
	gjUp0BjV+qJ3EkE94EEuPDjsIkWIngloJDl+WnAPEumZbfOTNOMQqNI/Yr8Gqi3boKd3TDWYPQQNS
	1DaRrXjw==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwln4-0001jY-Qv; Mon, 03 Sep 2018 10:09:02 +0000
Received: from merlin.infradead.org ([2001:8b0:10b:1231::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgk-0006by-IC
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:02:30 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=merlin.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=sz99Ed4cFqHyXgEVj3BM7faY7Awz0qAK84Sc1hbiz1k=; b=PuCWZOVRQ8ZZe8Trn3MZ7awio
 TXK8QDi6I41wm3w7w6w4hoZo3+nwt24ncDFAqWnjXhlAvgjRRVfl/0rJ2DvrjXYi/ZAQaqP6ihNZY
 tGcSSytmkGgIbv68omCSdYYN4AT5UlLtfXFnEPTQxKiz2gTaeuMEc14G9pen58N7gu3+iKYnAppeN
 cPPVRhUSYYjfrkmAO5F3B8Dund59HVIA3pSMpdPF5kN0v+6rHBwAVJJU+aac7EGoXOWMcLhcCojMd
 yop+iSCMxvLbq0BRpVBTiIyuhHU5A7G6fQkdmpcx0dtYZ63wQ5DFZBE84fZByc92zqMXZDWlzKrFT
 X4kSocXWg==;
Received: from mail-wm0-x243.google.com ([2a00:1450:400c:c09::243])
 by merlin.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgi-00071G-0e
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:28 +0000
Received: by mail-wm0-x243.google.com with SMTP id c14-v6so538700wmb.4
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:17 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=sz99Ed4cFqHyXgEVj3BM7faY7Awz0qAK84Sc1hbiz1k=;
 b=S2/ZIJCiEXJjhxXnVeHi+pLSw3LwTTWpDMaaNjklWtYTVm10+efVbe3sykAGxVbuQP
 gAaQ48ck4Rrqr1UW7OKRa967JWbD49cFo68ykrQqdtfsmRzV7tZJocaoENvFH9Ab+wGJ
 chuI3cCXIDznKsrmTZL0jKArWuOFHM8/euRo7yXtDF6lVM/J2YEjprSNIwSuporMrpnV
 765qGuaQX6PKlRgRD8Ns07+z1z3UamALBztNeJbmY7Cn9IWlr3mK/9iq3TOobGpwqamj
 C6IQwa1Gt2h0nFKW6rGW/k7aw7qJarrJgTZqOrd6QMxWW9thgY7lPRgQDlQjfmQCYR61
 s2Vw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=sz99Ed4cFqHyXgEVj3BM7faY7Awz0qAK84Sc1hbiz1k=;
 b=qOkukli5JAW+f09ve54+b/VVhbzUgX6yHOjjZqaBksy8Mrquys3qW6vNcY6g9M8iUY
 U3tKIcXe9lpuoWH/tnN87zGXTQA5HsPgFEy+JOTN0tfw1Cl2NRjZWcnCxugcIr0EtyD+
 iwpjBfoNSFi0vfFWU/V4khdbZFcgG0siLECPnv2IOW/IYwi3Og2X7pnJAVB5VJ4ineZ4
 wgJYZT9ubZmpwAyeLa2aC+GxaEnlcxFIKcL1vtCSOIoyOCAzdEau7QOUG774DC25g3mE
 +Sq2IHysx8aQfTs2bJ4omTrXj8JuSIm3C9q/PaXMZLgvIO+MfmoVrMwOYZMzVVEkZOiv
 H7+g==
X-Gm-Message-State: APzg51D/Et5T3GN8zzdbR5FassOFgLwYKc/mFeg7OaNuBnwDsR2119Vj
 MOaGZqu1Kn9ftJkggxWb6e0=
X-Google-Smtp-Source: 
 ANB0VdbnQ4DGqks37mn1q9DRTmofyTEpSofTDMg1HggzCg2Xp6sbNUIYPT285kUa1RnATH0fOvj9aw==
X-Received: by 2002:a1c:1805:: with SMTP id 5-v6mr4397863wmy.25.1535968935782;
 Mon, 03 Sep 2018 03:02:15 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.14
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:15 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 7/9] dt-bindings: ata: update ahci_sunxi bindings
Date: Mon,  3 Sep 2018 12:01:59 +0200
Message-Id: <20180903100201.23131-8-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_060228_064155_BD84D1FB 
X-CRM114-Status: GOOD (  13.01  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch update binding with the new R40 compatible.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
Reviewed-by: Rob Herring <robh@kernel.org>
---
 Documentation/devicetree/bindings/ata/ahci-platform.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/devicetree/bindings/ata/ahci-platform.txt b/Documentation/devicetree/bindings/ata/ahci-platform.txt
index 45b451961612..e30fd106df4f 100644
--- a/Documentation/devicetree/bindings/ata/ahci-platform.txt
+++ b/Documentation/devicetree/bindings/ata/ahci-platform.txt
@@ -10,6 +10,7 @@ PHYs.
 Required properties:
 - compatible        : compatible string, one of:
   - "allwinner,sun4i-a10-ahci"
+  - "allwinner,sun8i-r40-ahci"
   - "brcm,iproc-ahci"
   - "hisilicon,hisi-ahci"
   - "cavium,octeon-7130-ahci"
@@ -44,6 +45,7 @@ Required properties when using sub-nodes:
 - #address-cells    : number of cells to encode an address
 - #size-cells       : number of cells representing the size of an address
 
+For allwinner,sun8i-r40-ahci, the reset propertie must be present.
 
 Sub-nodes required properties:
 - reg		    : the port number
From patchwork Mon Sep  3 10:02:00 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585727
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 54DA516B1
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:17:49 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 3BC5C29250
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:17:49 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 2D17D2945E; Mon,  3 Sep 2018 10:17:49 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id BE06729250
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:17:48 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=XPGzgOzU9ZIpPi9w4jel9Hq+mwD/dfE3udf/nEwoAK0=; b=GQYnfUdPjVTOwEZlWEMEWMqYWY
	/VnAA8slKHkQC/Ov8AbSw9K1uENMEVcCfaOMhBusPWjJOBltQ2GovSSXuh8Xv/fqQiyDbUfPZhTG1
	Inw71scGg5yssHkGYd+3KcGPDdSn5e9bvKyjppSpEzmJ3jCNsrohffsZjP1FUiyt4pEyC20fJ5iEC
	RyERYKW5gqlVl6rAKsEfbUK8jMSMsbvtkpe0TMSL3I9zlrftpPRQ7CX+v94TLrZqAMGcFNom5ofRy
	NkFyf8194cEHnMA1nfsBn8VZcC27Jlx5G812laaWJnJFdFtS6n4dgqpJhtvoVDdQyr8VPzHIlUPq8
	nzbpxwNQ==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlvP-0007Vz-2f; Mon, 03 Sep 2018 10:17:39 +0000
Received: from casper.infradead.org ([2001:8b0:10b:1236::1])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlvN-0007UZ-EL
 for linux-arm-kernel@bombadil.infradead.org; Mon, 03 Sep 2018 10:17:37 +0000
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=infradead.org; s=casper.20170209; h=References:In-Reply-To:Message-Id:Date:
 Subject:Cc:To:From:Sender:Reply-To:MIME-Version:Content-Type:
 Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Id:
 List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
 bh=UNiJvsSjjdgkQ1ZQoChwHdQ+/Jgt1wIrgsMDUlZYcDA=; b=R+hej3Qzr6g7slXA9b96pQ1dn
 9+XcS9F/uTvD9kvN7BRaAVgsQKlQzuQyWolWKnvE4+qSQKj6bC4ZLGZhnzLe/h3kbGaPicJCp97Qr
 p3JC/hd3FCFqp6BB6Y6UebAeDFvZi1mnqbbxoiQgqStfya/xUPxJPgxWnw61tqtlUo/cpp04ZhIAr
 HacDYjR/7CuI6sZhVY2/vLppdzVk5CEtD13AXJTFAwyDyYCt17BrP3zQmZ4dej91sBUHeIHNpdhtP
 WNKYFMtuN+0syQLxT0Xvnjo6l9lG3BdMBIHyEp2K7rU4cYPOGNi/cO/F6o5qgaPfD441KKf6X/kYH
 keuUEqJSQ==;
Received: from mail-wr1-x444.google.com ([2a00:1450:4864:20::444])
 by casper.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgi-0000lS-KI
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:30 +0000
Received: by mail-wr1-x444.google.com with SMTP id u12-v6so34781wrr.4
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:18 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=UNiJvsSjjdgkQ1ZQoChwHdQ+/Jgt1wIrgsMDUlZYcDA=;
 b=Da0LDtvUx2ZakB3RYUo5s50VYBxKrxaxV+rN/HJEZP6SHYGGVcSrxGBYkVVFuMkMGK
 +dZ4PFxRXHuinvFrz2NSHlL6k/o/UB/YrXcn4QG1DJkvyuwRfX2bN1vablMwBV49QSyj
 XwjLVLlKhF6uFG0A5JCkMvIy1O4/XiJ0o7eFc3J5GCqn578gGR+Fr+LINYLX3IRHVoNX
 KN20NGujdnZC8PY21heiL2hKIOMpmjFNSml33FC71sQNXceZPqA6zjckHFG47mEriDqC
 dMiRnnrUizmsNroMIjSt4wFl4jBng3f7bYzftyzYnSe9tO6zYFr1hD0qzSGUkx7rBcJv
 yoIw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=UNiJvsSjjdgkQ1ZQoChwHdQ+/Jgt1wIrgsMDUlZYcDA=;
 b=h6/LnpfWKmTysfF5r4FKyekZIZLe+y1p5j3tpkU+b+LvQo2bJcL9P3/p5W0XxUTMWQ
 ApVQalr1MkXsZ0vZb57SaFo9SdTma7fkNo0ONkPrz1hdhIVjjuD7ywh9l38gfVsXflx0
 kYEgrJrE1qng6SBb6S1ZRtgegAap5OdP2UA/tzh4dI4+IP7/cHk1NKePBPcgDeQro/3A
 7w/x1Z0e4brftjZXAqjjqhi4xMrPqHEawByug6ApLlcgNVkRxKxqkIfrpqoka37nKn/p
 GVJW3AEYSSMf6S5P6qBJs1yql9INor1wdVTrnZPcVEkNC4jZNWBlQsuUsmplyIgkm/ru
 23Lg==
X-Gm-Message-State: APzg51BdID0oSNuRmjnNf4yc20sSB3fe5HdiinelUmilwG70TzPZP6kt
 XNzXcdKWWVy2COB0QKCzISU=
X-Google-Smtp-Source: 
 ANB0VdaNyNMsh0OqKfQXULzzxhfM4EIKfWqEDJ0m/NjlNsFiVZriZMTIOdT+FGXBpQeEF34mD+zj4Q==
X-Received: by 2002:adf:e792:: with SMTP id
 n18-v6mr7407609wrm.136.1535968937245;
 Mon, 03 Sep 2018 03:02:17 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.15
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:16 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 8/9] ARM: dts: sun8i: r40: add sata node
Date: Mon,  3 Sep 2018 12:02:00 +0200
Message-Id: <20180903100201.23131-9-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_110228_713785_A42A4DDA 
X-CRM114-Status: GOOD (  14.74  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org, Icenowy Zheng <icenowy@aosc.io>
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

R40 have a sata controller which is the same as A20.
This patch adds a DT node for it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 arch/arm/boot/dts/sun8i-r40.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-r40.dtsi b/arch/arm/boot/dts/sun8i-r40.dtsi
index 852c2ccc3268..d27c522e1918 100644
--- a/arch/arm/boot/dts/sun8i-r40.dtsi
+++ b/arch/arm/boot/dts/sun8i-r40.dtsi
@@ -550,6 +550,19 @@
 			#size-cells = <0>;
 		};
 
+		ahci: sata@1c18000 {
+			compatible = "allwinner,sun8i-r40-ahci";
+			reg = <0x01c18000 0x1000>;
+			interrupts = <GIC_SPI 56 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&ccu CLK_BUS_SATA>, <&ccu CLK_SATA>;
+			resets = <&ccu RST_BUS_SATA>;
+			resets-name = "ahci";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+
+		};
+
 		gmac: ethernet@1c50000 {
 			compatible = "allwinner,sun8i-r40-gmac";
 			syscon = <&ccu>;
From patchwork Mon Sep  3 10:02:01 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Corentin Labbe <clabbe.montjoie@gmail.com>
X-Patchwork-Id: 10585715
Return-Path: 
 <linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 580285A4
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:11:46 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 3F01329560
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:11:46 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 3226B29567; Mon,  3 Sep 2018 10:11:46 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-2.9 required=2.0 tests=BAYES_00,DKIM_ADSP_CUSTOM_MED,
	DKIM_SIGNED,DKIM_VALID,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_NONE
	autolearn=ham version=3.3.1
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id CF73B29560
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 10:11:45 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:MIME-Version:Cc:List-Subscribe:
	List-Help:List-Post:List-Archive:List-Unsubscribe:List-Id:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=qZuKYntMqeib1nr8fHoW97FU90zV/9d19nJgxtoyhNk=; b=MWOhLSVJUhBOuLm48JEq9nJu/Q
	KX9XgS6Gy1Dt9UnOtUshq+jjJp7sIoCIYBUXEUuSie8WOrCWi4JNGhHq/mqDYWwrZSlgk7G1CGlO8
	s44WZnLCcAEEBIMV5M8m8896CZZDo+kLSbA/jszjKcj5DTrl0G3B3NLeTssu7Qh1mr6qNfvMLcUjX
	dp7u/rVJ9CWjmBXFiBb+9DpEvaU07z0Mu9MUhFll/rBqcLRhQfBkYSGgC0FJ76rKSK9wWsQzvjFM0
	91+3Twcr1n4mwqusQNGDBAHyLQcCEUzA3XiixmQxWKpRLhNn0T4yWIO3VBLpEOc+MnZPzDQdTUci0
	w6+XNgsg==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.90_1 #2 (Red Hat Linux))
	id 1fwlpU-000416-Gc; Mon, 03 Sep 2018 10:11:32 +0000
Received: from mail-wr1-f65.google.com ([209.85.221.65])
 by bombadil.infradead.org with esmtps (Exim 4.90_1 #2 (Red Hat Linux))
 id 1fwlgk-0006YD-OL
 for linux-arm-kernel@lists.infradead.org; Mon, 03 Sep 2018 10:02:53 +0000
Received: by mail-wr1-f65.google.com with SMTP id o37-v6so20923wrf.6
 for <linux-arm-kernel@lists.infradead.org>;
 Mon, 03 Sep 2018 03:02:20 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=Yco3GuOwe1ADSho8sukxLRIOHQlCReJ4bgpdQLYejFc=;
 b=J1X5KvpQE8zgQgmc2/vQAmiKvPlYH1gGdMjr9Zuw9Y4Dos/aMtoSMHUNUhFqFXqeGd
 T3F8YIahgkY940DbwdulOsuINGsKtknyxXi1kvQpljEe4rM7fat35dAT+SWAt8J9vV3s
 7KVWYEmTaWvqHrjvfPUcKYav5YtxbOqh7/w+qM3qI4U6VrnLNjJsgzr95uDgM2KQJ6k0
 cnlooSaxrl5xHSkxW1DnZ73wVmzOjRbFWDo00o53FjBpoNaTQjix3YcAuGYbzqZZKtgn
 O7JBlpXiDHNSvGp+kmub0iow0Oi3DckfbwIeZxwSqJVqHMXDPObNVbpbl3azYMdcCA0h
 4bpg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=Yco3GuOwe1ADSho8sukxLRIOHQlCReJ4bgpdQLYejFc=;
 b=m5BHitpgv/8p9xVVv3Hqxmc91rCPiS80gopKu1NOAmbt18ysCfUy2b6ZzIJPetz5lt
 REiTibd6/pY1cHqBj8OdRlsoczFcfnc8GjRt1FwJQ2eUe8JpWbe+rMGyxQ60hjWpJ5a6
 Q0ZMM4pEhpRugSuypf+UrwI3UW6kEIfNphLfvxvRAqOoMcynOnZ4+Rge8fJVNScMlPI7
 H/zPRAMfcU5nAi/93Mq7LVmCrY6KEsCaI9vEPD5I9SeOqnOGuYgDaE+VkEi1KtlMc6Rn
 VOb9guw1dl4sOdXqos3CKfjq/Q0qOJN9nbA3mfGCgziAQnyWzxy4DjQ6qGx+LnFkeNND
 engQ==
X-Gm-Message-State: APzg51AKJCOKbj6AydDMx7qeWccgPV18G4AWIseNN2i5FJg5RgXUm+fU
 nN3LJwP9RcATgR+XeWe9QTQ=
X-Google-Smtp-Source: 
 ANB0VdYcw7DcZk2D5RcjweU2/i4DeLxR20feymZy2BW1uejbAnCva4EEXUo8FY/IQzBQh93UZ3UD+w==
X-Received: by 2002:adf:e084:: with SMTP id
 c4-v6mr17993852wri.199.1535968938709;
 Mon, 03 Sep 2018 03:02:18 -0700 (PDT)
Received: from Red.localdomain ([2a01:cb1d:147:7200:2e56:dcff:fed2:c6d6])
 by smtp.googlemail.com with ESMTPSA id
 k34-v6sm31773936wre.18.2018.09.03.03.02.17
 (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
 Mon, 03 Sep 2018 03:02:17 -0700 (PDT)
From: Corentin Labbe <clabbe.montjoie@gmail.com>
To: axboe@kernel.dk, hdegoede@redhat.com, mark.rutland@arm.com,
 maxime.ripard@bootlin.com, robh+dt@kernel.org, wens@csie.org
Subject: [PATCH v5 9/9] ARM: dts: sun8i: sun8i-r40-bananapi-m2-ultra: enable
 AHCI
Date: Mon,  3 Sep 2018 12:02:01 +0200
Message-Id: <20180903100201.23131-10-clabbe.montjoie@gmail.com>
X-Mailer: git-send-email 2.16.4
In-Reply-To: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
References: <20180903100201.23131-1-clabbe.montjoie@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20180903_030230_850938_DFF3B7DA 
X-CRM114-Status: GOOD (  12.29  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-ide@vger.kernel.org, linux-sunxi@googlegroups.com,
 Corentin Labbe <clabbe.montjoie@gmail.com>,
 linux-arm-kernel@lists.infradead.org
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch enable the AHCI controller.
Since this controller need two regulator, this patch add them.

Signed-off-by: Corentin Labbe <clabbe.montjoie@gmail.com>
---
 arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts b/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts
index a891a387e8f1..438b7b44dab3 100644
--- a/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts
+++ b/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts
@@ -105,6 +105,12 @@
 	};
 };
 
+&ahci {
+	ahci-supply = <&reg_dldo4>;
+	phy-supply = <&reg_eldo3>;
+	status = "okay";
+};
+
 &de {
 	status = "okay";
 };
@@ -250,6 +256,18 @@
 	regulator-name = "vcc-wifi";
 };
 
+&reg_dldo4 {
+	regulator-min-microvolt = <2500000>;
+	regulator-max-microvolt = <2500000>;
+	regulator-name = "vdd2v5-sata";
+};
+
+&reg_eldo3 {
+	regulator-min-microvolt = <1200000>;
+	regulator-max-microvolt = <1200000>;
+	regulator-name = "vdd1v2-sata";
+};
+
 &tcon_tv0 {
 	status = "okay";
 };
